src/erlcloud_sm.erl

```diff
@@ -0,0 +1,136 @@
+-module(erlcloud_sm).
+-author("joshua@halloapp.com").
+
+-include("erlcloud.hrl").
+-include("erlcloud_aws.hrl").
+
+%%% Library initialization.
+-export([new/2, new/3, new/4]).
+
+%%% API
+-export([
+    get_secret_value/2, get_secret_value/3
+]).
+
+%%%------------------------------------------------------------------------------
+%%% Shared types
+%%%------------------------------------------------------------------------------
+
+-type return_type() :: {ok, proplists:proplist()} | {error, term()}.
```

Comment:
I would suggest just calling this `sm_response()` or similar; `return_type()` seems more like an option that says what type of return you want, if that makes sense.